encoding/test_utils.go

```diff
@@ -0,0 +1,18 @@
+package encoding
+
+import (
+	"math"
+
+	"golang.org/x/exp/constraints"
+)
+
+// GeneratePowersOfTwo creates a slice of integers, containing powers of 2 (starting with element == 1), with
+// powersToGenerate number of elements
+func GeneratePowersOfTwo[T constraints.Integer](powersToGenerate T) []T {
+	powers := make([]T, powersToGenerate)
+	for i := T(0); i < powersToGenerate; i++ {
+		powers[i] = T(math.Pow(2, float64(i)))
+	}
+
+	return powers
```

Comment:
you can get more efficiently get powers of two by doing the following:
```suggestion
	powers := make([]T, powersToGenerate)
	for i := T(0); i < powersToGenerate; i++ {
		powers[i] = 1 << i
	}

	return powers
```

Reply:
Fixed